tests/test_keras_mixin.py

```diff
@@ -0,0 +1,153 @@
+import os
+import shutil
+import time
+import unittest
+
+from huggingface_hub import HfApi
+from huggingface_hub.file_download import is_tf_available
+from huggingface_hub.keras_mixin import KerasModelHubMixin
+
+from .testing_constants import ENDPOINT_STAGING, PASS, USER
+from .testing_utils import set_write_permission_and_retry
+
+
+REPO_NAME = "mixin-repo-{}".format(int(time.time() * 10e3))
+
+WORKING_REPO_SUBDIR = "fixtures/working_repo_3"
+WORKING_REPO_DIR = os.path.join(
+    os.path.dirname(os.path.abspath(__file__)), WORKING_REPO_SUBDIR
+)
+
+if is_tf_available():
+    import tensorflow as tf
+
+
+def require_tf(test_case):
+    """
+    Decorator marking a test that requires PyTorch.
+
+    These tests are skipped when PyTorch isn't installed.
+
+    """
+    if not is_tf_available():
+        return unittest.skip("test requires Tensorflow")(test_case)
+    else:
+        return test_case
+
+
+if is_tf_available():
+
+    class DummyModel(tf.keras.Model, KerasModelHubMixin):
+        def __init__(self, **kwargs):
+            super().__init__()
+            self.config = kwargs.pop("config", None)
+            self.l1 = tf.keras.layers.Dense(2, activation="relu")
+            dummy_batch_size = input_dim = 2
+            self.dummy_inputs = tf.ones([dummy_batch_size, input_dim])
+
+        def call(self, x):
+            return self.l1(x)
+
+
+else:
+    DummyModel = None
+
+
+@require_tf
+class HubMixingCommonTest(unittest.TestCase):
+    _api = HfApi(endpoint=ENDPOINT_STAGING)
```

Comment:
Is this super class really needed? We only have one test class so maybe the API wrapper should be instantiated in setUpClass?

Reply:
You're probably right. I didn't want to mess with it too much, though. again, just following the previous file. I did find the superclass useful (if I remember right) when I played around with parametrizing the tests so we could run them over TF/PyTorch/etc.